protocol/javatests/org/openyolo/protocol/AdditionalPropertiesTest.java

```diff
@@ -0,0 +1,222 @@
+                                    CredentialDeleteResult.CODE_DELETED))
+                    .put("CredentialRetrieveRequest",
+                            () -> new CredentialRetrieveRequest.Builder(
+                                    AuthenticationMethods.EMAIL))
+                    .put("CredentialRetrieveResult",
+                            () -> new CredentialRetrieveResult.Builder(
+                                    CredentialRetrieveResult.CODE_NO_CREDENTIALS_AVAILABLE))
+                    .put("CredentialSaveRequest",
+                            () -> new CredentialSaveRequest.Builder(
+                                    ValidFacebookCredential.make()))
+                    .put("CredentialSaveResult",
+                            () -> new CredentialSaveResult.Builder(
+                                    CredentialSaveResult.CODE_SAVED))
+                    .put("Hint",
+                            () -> new Hint.Builder(
+                                    ValidFacebookCredential.ID,
+                                    ValidFacebookCredential.AUTHENTICATION_METHOD))
+                    .put("HintRetrieveRequest",
+                            () -> new HintRetrieveRequest.Builder(
+                                    AuthenticationMethods.EMAIL))
+                    .put("HintRetrieveResult",
+                            () -> new HintRetrieveResult.Builder(
+                                    HintRetrieveResult.CODE_NO_HINTS_AVAILABLE))
+            .build();
+
+    @Parameters(name = "AdditionalPropertiesTest for {0}")
+    public static Collection params() {
+        ArrayList<Object[]> params = new ArrayList<>();
+
+        for (String containerTypeName : buildersByContainerType.keySet()) {
+            params.add(new Object[] { containerTypeName });
+        }
+
+        return params;
+    }
+
+    private AdditionalPropertiesBuilder<?, ?> mBuilder;
+
+    public AdditionalPropertiesTest(String containerClassName) {
+        mContainerClassName = containerClassName;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        mBuilder = buildersByContainerType.get(mContainerClassName).call();
+    }
+
+    @Test
+    public void testBuilder_setAdditionalProperties() {
+        AdditionalPropertiesContainer container = mBuilder
+                .setAdditionalProperties(ValidAdditionalProperties.make())
+                .build();
+        ValidAdditionalProperties.assertEquals(container.getAdditionalProperties());
+    }
+
+    @Test
+    public void testBuilder_setAdditionalProperties_withNull() {
+        AdditionalPropertiesContainer container = mBuilder
+                .setAdditionalProperties(ValidAdditionalProperties.make())
+                .setAdditionalProperties(null)
+                .build();
+
+        assertThat(container.getAdditionalProperties()).isEmpty();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
1) Nit: Use mBuilder instead?
2) Nit: @Test(expected = ...) could catch an exception from other code.  Something like assertThrows is preferred, but AFAIK isn't in Junit4.x. I did find http://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.htm which seems similar.
3) Nit: Remove .build()

Reply:
I think assertThrows is available in the assertion library we're using, but expects Java 8. That's actually fine now that we're using the latest build tools, so I can try that out in a follow-up.

Reply:
And yeah, the mBuilder thing isn't a nit, it means I'm testing the wrong thing :) Good catch.